linalg/amgxsolver.hpp

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+//Reference:
+//Pi-Yueh Chuang, & Lorena A. Barba (2017).
+//AmgXWrapper: An interface between PETSc and the NVIDIA AmgX library. J. Open Source Software, 2(16):280, doi:10.21105/joss.00280
+
+#ifndef MFEM_AMGX_SOLVER
+#define MFEM_AMGX_SOLVER
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_AMGX
+
+
+#include <amgx_c.h>
+#ifdef MFEM_USE_MPI
+#include <mpi.h>
+#include "hypre.hpp"
+#else
+#include "operator.hpp"
+#include "sparsemat.hpp"
+#endif
+
+namespace mfem
+{
+
+class AmgXSolver : public Solver
+{
+
+public:
+
+   AmgXSolver() = default;
+
+   /* Constructor for serial builds - supported without Hypre and MPI */
+   AmgXSolver(const std::string &cfgFile);
```

Comment:
As other reviewers have suggested, I think an interface that does not require a configuration file would be useful. We can follow roughly the thinking behind `HypreBoomerAMG`, that is, unless you specify otherwise it configures as a preconditioner that does one V-cycle with some standard options.